packages/imperative/src/imperative/src/plugins/utilities/npm-interface/install.ts

```diff
@@ -134,14 +139,85 @@ export async function install(packageLocation: string, registry: string, install
+                        // Returns true if the object was updated, and false otherwise
+                        const updateExtendersJson = (profile: IProfileTypeConfiguration): boolean => {
+                            if (!(profile.type in extendersJson.profileTypes)) {
+                                // If the type doesn't exist, add it to extenders.json and return
+                                extendersJson.profileTypes[profile.type] = {
+                                    from: [packageInfo.name],
+                                    version: profile.schemaVersion
+                                };
+                                return true;
+                            }
+
+                            // Otherwise, only update extenders.json if the schema version is newer
+                            const existingTypeInfo = extendersJson.profileTypes[profile.type];
+                            if (semver.valid(existingTypeInfo.version)) {
+                                if (profile.schemaVersion && semver.lt(profile.schemaVersion, existingTypeInfo.version)) {
+                                    return false;
+                                }
+                            }
+
+                            extendersJson.profileTypes[profile.type] = {
+                                from: [packageInfo.name],
+                                version: profile.schemaVersion
+                            };
+                            return true;
+                        };
+
+                        // Determine new profile types to add to schema
+                        let shouldUpdate = false;
+                        for (const profile of pluginImpConfig.profiles) {
+                            if (!existingTypes.includes(profile.type)) {
+                                loadedSchema.push(profile);
+                            } else {
+                                const existingType = loadedSchema.find((obj) => obj.type === profile.type);
+                                if (semver.valid(existingType.schemaVersion)) {
+                                    if (semver.gt(profile.schemaVersion, existingType.schemaVersion)) {
+                                        existingType.schema = profile.schema;
+                                        existingType.schemaVersion = profile.schemaVersion;
+                                    }
+                                } else {
+                                    existingType.schema = profile.schema;
+                                    existingType.schemaVersion = profile.schemaVersion;
+                                }
+                            }
+                            shouldUpdate = shouldUpdate || updateExtendersJson(profile);
+                        }
+
+                        if (shouldUpdate) {
+                            // Update extenders.json (if necessary) after installing the plugin
+                            ProfileInfo.writeExtendersJson(extendersJson);
```

Comment:
This seems like a clever way to loop through all profile types and make sure we update the `extendersJson` object.

However, after the first time that the `extenderJson` objectis updated, the `updateExtendersJson` method won't ever be called again.

I might be reading things wrong, but don't we need to call the `updateExtendersJson` for every profile type?

Otherwise, we might be writing an incomplete `extender.json` 😋

Reply:
Ah, good catch! That's correct, I overlooked the OR short-circuiting that could happen with `shouldUpdate`. I think if I set it to `updateExtendersJson(profile) || shouldUpdate`, that should ensure that `updateExtendersJson` is updated for every type.

I will add a couple unit tests to this PR as well to ensure that all the types are added if multiple are provided 👍

Reply:
Resolved in [13c1e94](https://github.com/zowe/zowe-cli/pull/2015/commits/13c1e94132a75fd7e50e0be322891d251d59b62b) 🙂

Reply:
Resolved in 13c1e94